primitives/src/signatures/mod.rs

```diff
@@ -0,0 +1,58 @@
+//! Module for signature primitives.
+
+use crate::errors::PrimitivesError;
+use ark_std::rand::{CryptoRng, RngCore};
+
+pub mod bls;
+pub mod schnorr;
+
+/// Trait definition for a signature scheme.
+// A signature scheme is associated with a hash function H that is
+// to be used for challenge generation.
+// FIXME: update H bound once hash-api is merged.
+pub trait SignatureScheme<H> {
```

Comment:
curious, can our current implementation fit into [`trait SignatureScheme` by arkwork](https://docs.rs/ark-crypto-primitives/latest/ark_crypto_primitives/signature/trait.SignatureScheme.html)?

it's kinda weird to see a general Signature Scheme trait generic over a choice of hash.

Reply:
Second thought. I think I misunderstood your point. The trait itself does not need to associate with a hash function. We should associate hash function with the actual instantiation.

Reply:
[RustCrypto's trait](https://docs.rs/signature/latest/signature/)